### PR TITLE
Add APP_KEYS environment variable to .environment

### LINKED ISSUE #178: Strapi: `project:init` return the wrong .environment file
Hi @akalipetis 

I start writing the Upsun Strapi guide and i needed to change all of the envVars in it as it is not well design for Strapi.
This is what i get: 
```
export RELATIONSHIPS_JSON="$(echo $PLATFORM_RELATIONSHIPS | base64 --decode)"

# Set database environment variables
export DB_HOSTNAME="$POSTGRESQL_HOSTNAME"
export DB_PORT="$POSTGRESQL_PORT"
export DB_PATH="$POSTGRESQL_PATH"
export DB_USERNAME="$POSTGRESQL_USERNAME"
export DB_PASSWORD="$POSTGRESQL_PASSWORD"
export DB_SCHEME="$POSTGRESQL_SCHEME"
export DATABASE_URL="${DB_SCHEME}://${DB_USERNAME}:${DB_PASSWORD}@${DB_HOSTNAME}:${DB_PORT}/${DB_PATH}"

# Strapi-specific environment variables
export JWT_SECRET="$PLATFORM_PROJECT_ENTROPY"
export API_TOKEN_SALT="$PLATFORM_PROJECT_ENTROPY"
```

And this is what is working to install it on Upsun:
```
export RELATIONSHIPS_JSON="$(echo $PLATFORM_RELATIONSHIPS | base64 --decode)"

# Set database environment variables
export DATABASE_HOSTNAME="$POSTGRESQL_HOSTNAME"
export DATABASE_HOST="$POSTGRESQL_HOST"                        --> missing one, required by Strapi
export DATABASE_PORT="$POSTGRESQL_PORT"
export DATABASE_PATH="$POSTGRESQL_PATH"
export DATABASE_USERNAME="$POSTGRESQL_USERNAME"
export DATABASE_PASSWORD="$POSTGRESQL_PASSWORD"
export DATABASE_SCHEME="$POSTGRESQL_SCHEME"
export DATABASE_URL="${DATABASE_SCHEME}://${DATABASE_USERNAME}:${DATABASE_PASSWORD}@${DATABASE_HOSTNAME}:${DATABASE_PORT}/${DATABASE_PATH}"

# Strapi-specific environment variables
export JWT_SECRET="$PLATFORM_PROJECT_ENTROPY"
export ADMIN_JWT_SECRET="$PLATFORM_PROJECT_ENTROPY"        --> missing one, required by Strapi
export API_TOKEN_SALT="$PLATFORM_PROJECT_ENTROPY"
export APP_KEYS="toBeModified1,toBeModified2"                                    --> missing one, required by Strapi
```

- envVar needs to start with `DATABASE_` as used in `database.js` 
- missing `DATABASE_HOST`
- missing `ADMIN_JWT_SECRET`
- missing `APP_KEYS`

You can have a look at the draft guide here:
https://docs.upsun.com.adds-strapi-guide-3o6f4xi-652soceglkw4u.eu-3.platformsh.site/get-started/strapi.html#customize-generated-environment-file

ps: and as for now, image NodeJs 18/20 is not generating the related service envVars in the container (`POSTGRESQL_*`) that's why in the guide i need to extract them one by one from `RELATIONSHIPS_JSON`

### PR DESCRIPTION
**Pull request by akalipetis**
Also, make sure all optional Strapi environment variables are not set, if they already exist in the environment. This allows for setting them as project sensitive environment variables.

It's a bit easier to review with [whitespaces disabled](https://github.com/platformsh/platformify/pull/189/files?w=1)

Fix #178

### CODE REVIEW

File: platformifier/templates/generic/.environment [hunk 1]
```diff
@@ -32,9 +32,20 @@ export DATABASE_PORT="$DB_PORT"
 export DATABASE_NAME="$DB_PATH"
 export DATABASE_USERNAME="$DB_USERNAME"
 export DATABASE_PASSWORD="$DB_PASSWORD"
-export DATABASE_SCHEME="$DB_SCHEME"
+export DATABASE_SCHEME="$DB_CONNECTION"
```
Comment (flovntp) [inline platformifier/templates/generic/.environment:35]: @akalipetis why did you change `DB_SCHEME` to `DB_CONNECTION` but keep `DB_SCHEME` in the `platformifier/templates/upsun/.environment` file ?

File: platformifier/templates/generic/.environment [hunk 2]
```diff
@@ -32,9 +32,20 @@ export DATABASE_PORT="$DB_PORT"
...
+  export ADMIN_JWT_SECRET="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
+fi
+if [[ -z "$JWT_SECRET" ]]; then
+  export JWT_SECRET="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
+fi
+if [[ -z "$API_TOKEN_SALT" ]]; then
+    export API_TOKEN_SALT="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
+fi
+if [[ -z "$APP_KEYS" ]]; then
+    export APP_KEYS="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
```
Comment (flovntp) [inline platformifier/templates/generic/.environment:49]: The `APP_KEYS` needs to have 2 random strings separated by a comma and here you just add 1 :( 
this should be: 
```
export APP_KEYS="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY,${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
```

Comment (flovntp) [reply platformifier/templates/generic/.environment:49]: as i'm not sure how to define 2 different random strings and as @gilzow is right saying that random string could be have the same value, you can duplicate the PROJECT_ENTROPY ;)

File: platformifier/templates/upsun/.environment [hunk 3]
```diff
@@ -36,7 +36,18 @@ export DATABASE_USERNAME="$DB_USERNAME"
...
+  export ADMIN_JWT_SECRET="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
+fi
+if [[ -z "$JWT_SECRET" ]]; then
+  export JWT_SECRET="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
+fi
+if [[ -z "$API_TOKEN_SALT" ]]; then
+    export API_TOKEN_SALT="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
+fi
+if [[ -z "$APP_KEYS" ]]; then
+    export APP_KEYS="${{ .Assets.EnvPrefix }}_PROJECT_ENTROPY"
```
Comment (flovntp) [inline platformifier/templates/upsun/.environment:51]: same comment as in the `generic/.environment` --> please double PROJECT_ENTROPY

File: platformifier/templates/generic/.environment (hunk 1 above)
Comment (akalipetis) [reply platformifier/templates/generic/.environment:35]: It's `DB_CONNECTION` in here - there was a typo previously.

File: platformifier/templates/generic/.environment (hunk 2 above)
Comment (akalipetis) [reply platformifier/templates/generic/.environment:49]: One key is required, multiple keys are used for rotating keys (the first key is always used for signing and any more keys are only used for verification, so that you can safely rotate the key and at some point decide to invalidate old keys).

You can read more about this in the following links:

* Koa `app.keys` (underlying framework): https://koajs.com/
* Keygrip (underlying library): https://github.com/crypto-utils/keygrip

> The keylist is an array of all valid keys for signing, in descending order of freshness; new keys should be unshifted into the array and old keys should be popped.

> keys.sign(data)
> This creates a SHA1 HMAC based on the first key in the keylist, and outputs it as a 27-byte url-safe base64 digest (base64 without padding, replacing + with - and / with _).

**Comment by akalipetis** [comment]
@flovntp I've replied to both comments, let me know if you're OK so that I can proceed with merging.

**Review by flovntp — APPROVED** [review]
